nixos/modules/services/networking/wireguard.nix

```diff
@@ -245,7 +245,9 @@ let
         path = with pkgs; [ kmod iproute wireguard-tools ];
 
         serviceConfig = {
-          Type = "oneshot";
+          Type = "simple";
+          Restart = "on-failure";
+          RestartSec = "5s";
```

Comment:
I think there was a restart limit in systemd or something, which could lead to the service not being restarted infinite.

Reply:
For completeness, yes, you need

```nix
      unitConfig = {
        StartLimitIntervalSec = 0; # ensure Restart= is always honoured
      };
```